src/test/java/com/google/sps/servlets/PotentialMatchesServletTest.java

```diff
@@ -35,16 +37,27 @@
 import org.mockito.MockitoAnnotations;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
 import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
 import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.common.collect.ImmutableSet;
 import com.google.sps.servlets.PotentialMatchesServlet;
+import com.google.sps.data.PotentialMatchAlgorithm;
+import com.google.sps.data.friend_map.UserFriendsMap;
+import com.google.sps.data.friend_map.UserNode;
 
 @RunWith(JUnit4.class)
 public class PotentialMatchesServletTest {
   private static final String NO_POTENTIAL_MATCH_RESULT = "NO_POTENTIAL_MATCHES";
   private static final String MATCHINFO_NEXT_MATCH_ID_FIELD = "nextPotentialMatchID";
   private static final String USER_ID_REQUEST_URL_PARAM = "userid";
+  private static final String MATCH_INFO_ENTITY = "match-info";
+  private static final String POTENTIAL_MATCHES_PROPERTY = "potential-matches";
+  private static final String FRIENDED_IDS_PROPERTY = "friended-ids";
+  private static final String PASSED_IDS_PROPERTY = "passed-ids";
```

Comment:
any particular reason these are duplicated here?

Reply:
The constants on line 54 through 58 are being used in the test cases to refer to certain parameter, entity, and property names and certain values. I'm noticing that I didn't use the last two properties, so I can remove these.

Reply:
Yeah, what I meant was that you could reference the constants already defined in PotentialMatchesServlet to avoid duplicating (after making them package-private).

Though sometimes tests *intentionally* duplicate values that should never be changed, so that the test will detect if someone *does* change them. These strings fall into that category, so what you have is also fine.